internal/services/tesla_fleet_api_service.go

```diff
@@ -91,50 +90,100 @@ func (t *teslaFleetAPIService) CompleteTeslaAuthCodeExchange(ctx context.Context
 	}, nil
 }
 
-// GetVehicles godoc
+// GetVehicles
 // @Description Call Tesla Fleet API to get a list of vehicles using authorization token
-// @Param       token - authorization token to be used as bearer token
-// @Param       region - API region which is used to determine which fleet api to call
-// @Returns     {array} []services.TeslaVehicle
 func (t *teslaFleetAPIService) GetVehicles(ctx context.Context, token, region string) ([]TeslaVehicle, error) {
 	baseURL := fmt.Sprintf(t.Settings.Tesla.FleetAPI, region)
 	url := baseURL + "/api/1/vehicles"
 
+	resp, err := t.performTeslaGetRequest(ctx, url, token)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch vehicles for user: %w", err)
+	}
+	defer resp.Body.Close()
+
+	vehicles := new(GetVehiclesResponse)
+	if err := json.NewDecoder(resp.Body).Decode(vehicles); err != nil {
+		return nil, fmt.Errorf("invalid response encountered while fetching user vehicles: %w", err)
+	}
+
+	if vehicles.Response == nil {
+		return nil, fmt.Errorf("error occurred fetching user vehicles")
+	}
+
+	return vehicles.Response, nil
+}
+
+// GetVehicle
+// @Description Call Tesla Fleet API to get a single vehicle by ID
+func (t *teslaFleetAPIService) GetVehicle(ctx context.Context, token, region string, vehicleID int) (*TeslaVehicle, error) {
+	baseURL := fmt.Sprintf(t.Settings.Tesla.FleetAPI, region)
+	url := fmt.Sprintf("%s/api/1/vehicles/%d", baseURL, vehicleID)
+
+	resp, err := t.performTeslaGetRequest(ctx, url, token)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch vehicles for user: %w", err)
+	}
+	defer resp.Body.Close()
+
+	vehicle := new(struct {
+		Response TeslaVehicle `json:"response"`
+	})
```

Comment:
We can move this struct definition outside the function since we don't need to define the struct every time it is run.

Reply:
unresolving this because it wasn't addressed.

Reply:
https://github.com/DIMO-Network/devices-api/blob/88a934688599f3bdf8d3be47034b1de3fd1d67b3/internal/services/tesla_fleet_api_service.go#L130